hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/AbstractMessageTask.java

```diff
@@ -295,9 +294,16 @@ protected void sendClientMessage(ClientMessage resultClientMessage) {
                 buf.writeBytes(frame.content);
                 frame = frame.next;
             }
-            //nasty
-            upcast(buf.byteBuffer()).flip();
-            asyncSocket.writeAndFlush(buf);
+
+            buf.flip();
+            if (!asyncSocket.writeAndFlush(buf)) {
+                // Unlike the 'classic' networking, the asyncSocket has a bound on the
+                // number of packets on the write-queue to prevent running into OOME.
+                // So if the response can't be send, we close the connection to indicate
+                // that the connection was congested.
+                connection.close("Closing connection: response could not be send due to congested socket "
```

Comment:
```suggestion
                connection.close("Response could not be send due to congested socket "
```
I believe the first part will be printed by the connection.close, so no need to duplicate it